code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -329,35 +330,35 @@
 			var/datum/gas_mixture/gas = location.remove_air(0.25 * environment.total_moles())
 			if(!gas)
 				return
-			if(!regulating_temperature)
-				regulating_temperature = 1
+			if(!regulating_temperature && thermostat_state == TRUE)
+				regulating_temperature = TRUE
 				visible_message("\The [src] clicks as it starts [environment.temperature > target_temperature ? "cooling" : "heating"] the room.", "You hear a click and a faint electronic hum.")
 
 			if(target_temperature > MAX_TEMPERATURE)
 				target_temperature = MAX_TEMPERATURE
 
 			if(target_temperature < MIN_TEMPERATURE)
 				target_temperature = MIN_TEMPERATURE
+			if(thermostat_state == TRUE)
+				var/heat_capacity = gas.heat_capacity()
+				var/energy_used = max(abs(heat_capacity * (gas.temperature - target_temperature) ), MAX_ENERGY_CHANGE)
 
-			var/heat_capacity = gas.heat_capacity()
-			var/energy_used = max(abs(heat_capacity * (gas.temperature - target_temperature) ), MAX_ENERGY_CHANGE)
+				//Use power.  Assuming that each power unit represents 1000 watts....
+				use_power(energy_used / 1000, ENVIRON)
 
-			//Use power.  Assuming that each power unit represents 1000 watts....
-			use_power(energy_used/1000, ENVIRON)
+				//We need to cool ourselves.
+				if(heat_capacity)
+					if(environment.temperature > target_temperature)
+						gas.temperature -= energy_used / heat_capacity
+					else
+						gas.temperature += energy_used / heat_capacity
 
-			//We need to cool ourselves.
-			if(heat_capacity)
-				if(environment.temperature > target_temperature)
-					gas.temperature -= energy_used / heat_capacity
-				else
-					gas.temperature += energy_used / heat_capacity
+				if(abs(environment.temperature - target_temperature) <= 0.5)
+					regulating_temperature = FALSE
+					visible_message("\The [src] clicks quietly as it stops [environment.temperature > target_temperature ? "cooling" : "heating"] the room.", "You hear a click as a faint electronic humming stops.")
```

Comment:
Don't need a `\The` here. It was here before of course, but you're editing the area already so may as well clean it up a bit while you're at it.
```suggestion
					visible_message("[src] clicks quietly as it stops [environment.temperature > target_temperature ? "cooling" : "heating"] the room.", "You hear a click as a faint electronic humming stops.")
```